### PR TITLE
Floating insert menu keyboard navigation fixes

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
Tabbing in the floating insert menu to higlight the checkboxes would clear out the filter input text. Pressing enter while the checkboxes are higlighted would dismiss the dialog.

image
**Fix:**
I lifted the input text to be controlled state of the `FloatingMenu` instead of controlled internally by react-select. Then I needed a way to expose the current highlighted item, so I copied @Rheeseyb 's aria solution. Works loke a charm!

**Commit Details:**
- Switched to react-winowed-select to get the latest version to access the aria event stream
- new `activelySelectedInsertOptionRef` contains the highlighted insertion option
- new state `filterInputValue` that holds the input text typed by the user
- only call `setFilterInputValue` on legit input changes, do not allow clearing on `input-blur`
- `useHandleCloseOnESCOrEnter` now submits `onChange` on Enter.

### CODE REVIEW

File: editor/src/components/canvas/ui/floating-insert-menu.tsx [hunk 1]
```diff
@@ -109,7 +114,7 @@ function useGetInsertableComponents(): InsertableComponentFlatList {
   return insertableComponents
 }
 
-function useComponentSelectorStyles(): StylesConfig {
+function useComponentSelectorStyles(): StylesConfig<InsertMenuItem, false> {
```
Comment (balazsbajorics) [inline editor/src/components/canvas/ui/floating-insert-menu.tsx:117]: type improvement thanks to newer version of react-windowed-select

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/ui/floating-insert-menu.tsx [hunk 2]
```diff
@@ -174,10 +179,10 @@ function useComponentSelectorStyles(): StylesConfig {
         display: 'flex',
         paddingTop: 2,
         opacity: 0.4,
-        color: data.color,
+        color: styles.color,
```
Comment (Rheeseyb) [inline editor/src/components/canvas/ui/floating-insert-menu.tsx:182]: Ha oops?!

File: editor/src/components/canvas/ui/floating-insert-menu.tsx [hunk 3]
```diff
@@ -305,6 +310,27 @@ function getMenuTitle(insertMenuMode: 'closed' | 'insert' | 'convert' | 'wrap'):
...
+      onFocus: ({ focused }: { focused: InsertMenuItem }) => {
+        activelySelectedInsertOptionRef.current = focused
+      },
+    }),
+    [],
+  )
+
+  const [filterInputValue, setFilterInputValue] = React.useState('')
+  const onInputValueChange = React.useCallback((newValue, actionMeta: InputActionMeta) => {
+    // when the user "tabs out" to the checkboxes, prevent react-select from clearing the input text
```
Comment (Rheeseyb) [inline editor/src/components/canvas/ui/floating-insert-menu.tsx:328]: If the user decides to select something on the canvas or perform some other (non-keyboard) interaction that would close the context menu, will this result in us still retaining the previous input value?

File: editor/src/components/canvas/ui/floating-insert-menu.tsx (hunk 2 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/ui/floating-insert-menu.tsx:182]: yeah, thanks type improvement!

File: editor/src/components/canvas/ui/floating-insert-menu.tsx (hunk 3 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/ui/floating-insert-menu.tsx:328]: nope, there's two components, one is a generic holding container that renders FloatingPopup : null. so closing the popup actually unmounts the FloatingPopup component, erasing all the internal state we might have had

**Comment by balazsbajorics** [comment]
Rheese I'm gonna assume you approved this! ❤️

**Comment by Rheeseyb** [comment]
> Rheese I'm gonna assume you approved this!

Yeah I was just waiting on the answer to that question about closing it and maintaining the state. :+1:

**Pull request merged**